File: DesignPatternsExercises/java/FacadeDesignPattern/Exercise/BookCheckoutFacade.java
public class BookCheckoutFacade {

	private int studentId;
	private StudentAccountVerification studentAccountVerification;
	private BookAvailability bookAvailability;

	public BookCheckoutFacade(int studentId) {
		this.studentId = studentId;
		studentAccountVerification = __________(3)__________;
		bookAvailability = __________(4)__________;
	}

	public void __________(5)__________(String book) {
		if(studentAccountVerification.isStudentIdValid(studentId) && bookAvailability.isBookAvailable(book)) {
			bookAvailability.__________(6)__________(book);
		}
	}
}

File: DesignPatternsExercises/ruby/facade_design_pattern/exercise/book_checkout_facade.rb
require_relative 'book_availability.rb'
require_relative 'student_account_verification.rb'

class BookCheckoutFacade
  def initialize(student_id)
    @student_id = student_id
    @student_account_verification = __________(4)__________
    @book_availability = __________(5)__________
  end

  def issue_book(book)
    @book_availability.__________(6)__________(book) if
        @student_account_verification.student_id_valid?(@student_id) && @book_availability.book_available?(book)
  end
end
